pkg/loki/modules.go

```diff
@@ -277,6 +292,17 @@ func (t *Loki) stopTableManager() error {
 }
 
 func (t *Loki) initStore() (err error) {
+	if ActiveIndexType(t.cfg.SchemaConfig) == "boltdb" && t.cfg.StorageConfig.BoltDBArchiverConfig.Enable {
+		t.cfg.StorageConfig.BoltDBArchiverConfig.IngesterName = t.cfg.Ingester.LifecyclerConfig.ID
+		if t.cfg.Target == Ingester {
```

Comment:
How is this going to work in a single binary mode ? I think it would overrides one of them no ?

Reply:
Active boltdb files are kept in a separate directory anyways. This is just an optimization to avoid downloading files unnecessarily.

Reply:
I think you should use a `switch` here and include `"all"` target just for clarity of what is going on.

Reply:
My concern is we don't know really which one is picked for `all` probably none so default to `ArchiverModeReadWrite` ?

Reply:
Yes, it would be `ArchiverModeReadWrite`